homeassistant/components/home_connect/sensor.py

```diff
@@ -50,38 +55,153 @@ class HomeConnectSensorEntityDescription(SensorEntityDescription):
     appliance_types: tuple[str, ...]
 
 
-SENSORS: tuple[HomeConnectSensorEntityDescription, ...] = (
-    HomeConnectSensorEntityDescription(
+SENSORS = (
+    SensorEntityDescription(
+        key=BSH_DOOR_STATE,
+        has_entity_name=True,
+        options=[
+            "Closed",
+            "Locked",
+            "Open",
+        ],
+        device_class=SensorDeviceClass.ENUM,
+        translation_key=bsh_key_to_translation_key(BSH_DOOR_STATE),
+    ),
+    SensorEntityDescription(
+        key="Refrigeration.Common.Status.Door.BottleCooler",
+        has_entity_name=True,
+        options=[
+            "Closed",
+            "Open",
```

Comment:
Since it was only 2 states I ended up [switching these](https://github.com/home-assistant/core/pull/125490#discussion_r1754085889) to the `binary_sensor` platform in PR #125490 .

Reply:
It makes sense.
Given that the API docs specifies that door states use an enum data type, I believe enum sensors are a better choice.
Here’s why: if any of these entities, which we currently assume that they have only 2 states, suddenly support an additional state (like a locked state for doors, or any new possible state), we need to ensure compatibility. This would mean deprecating the binary door sensor and introducing a new enum sensor, along with handling any collateral effects.